codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/c2j/rest-json/service-2.json

```diff
@@ -210,7 +210,8 @@
       "output":{"shape":"ChecksumStructureWithStreaming"},
       "httpChecksum" : {
         "requestValidationModeMember": "ChecksumMode",
-        "responseAlgorithms": ["CRC32C", "CRC32", "SHA1", "SHA256"]
+        "responseAlgorithms": ["CRC32C", "CRC32", "SHA1", "SHA256"],
+        "requestAlgorithmMember": "ChecksumAlgorithm"
```

Comment:
Added `requestAlgorithmMember`  because per smithy doc https://smithy.io/2.0/aws/aws-core.html#aws-protocols-httpchecksum-trait, either `requestAlgorithmMember` or `requestChecksumRequired` is required

> The httpChecksum trait MUST define at least one of the request checksumming behavior, by setting the requestAlgorithmMember or requestChecksumRequired property,